### PR TITLE
fix(plugins/dates): validate days and months in `mm/dd` format

### PR DESCRIPTION
**Pull request by Fdawgs**
At present, compromise-dates tags anything `2-digit/2-digit` as a `Date`:

```js
'use strict'

const nlp = require('compromise')
const nlpDates = require('compromise-dates')
nlp.plugin(nlpDates)

const text = 'I saw him 20/24'

const doc = nlp(text)
console.log(doc.out('tags'))
/*
  Outputs:
  [
    {
      i: [ 'Noun', 'Pronoun' ],
      saw: [ 'Verb', 'PastTense' ],
      him: [ 'Noun', 'Pronoun' ],
      '20/24': [ 'Date' ]
    }
  ]
*/
```

This PR updates the compromise-dates plugin to perform some minor validation of days and months when provided in the `mm/dd` format, by ensuring the months are between 01-12 and the days are between 01-31.

### CODE REVIEW

**Comment by Fdawgs** [comment]
Tests are failing because `mac-os-latest` is now using ARM.
See https://github.com/fastify/.github/issues/37

**Comment by hchiam** [comment]
i like the idea of tightening up the date detection, but i wonder if some english documents ever have the day before the month but specifically _without_ the _year_, as in "dd/mm"? i can anecdotally say i've encountered "mm/dd" and "dd/mm/yy" etc on physical displays, but i'm having no luck finding exact sources online of counter-examples using "dd/mm"

**Comment by hchiam** [comment]
> i like the idea of tightening up the date detection, but i wonder if some english documents ever have the day before the month but specifically _without_ the _year_, as in `dd/mm`? i can anecdotally say i've encountered `mm/dd` and `dd/mm/yy` etc on physical displays and digital documents, but i'm having no luck finding exact sources online of counter-examples using `dd/mm`

just found an explicit description: https://www.grammarly.com/blog/how-to-write-dates/#:~:text=writing%20dates%20as%20numerals

> So remember, if you are American and you write to your British friend inviting him to celebrate Independence Day on 7/4 with you, you can expect your guest to arrive on April 7 (which he will express as 7 April). Likewise, if he invites you to his Guy Fawkes Day party on 5/11, you will need to mark your calendar for November 5 rather than May 11.

File: plugins/dates/src/model/regex.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ export default [
   // @4pm
   [/^@[1-9]+(am|pm)$/, 'Time', '@5pm'],
   // 03/02
-  [/^[0-9]{2}\/[0-9]{2}$/, 'Date', '03/02'],
+  [/^(?:0[1-9]|1[0-2])\/(?:0[1-9]|[12]\d|3[01])$/, 'Date', '03/02'],
```
Comment (hchiam) [inline plugins/dates/src/model/regex.js:13]: given this info (https://github.com/spencermountain/compromise/pull/1129#issuecomment-2248977259), maybe this:
```suggestion
  [/^(?:0[1-9]|[12]\d|3[01])\/(?:0[1-9]|[12]\d|3[01])$/, 'Date', '03/02'],
```

**Review by hchiam — CHANGES_REQUESTED** [review]

**Comment by Fdawgs** [comment]
@hchiam  British dates aren't fully supported as stated in the readme.

Example:
```js
const doc = nlp('I have to go to the hellhole that is London on 12/08').dates().get();

console.log(doc);
/*
  [{ start: '2024-12-08', end: '2024-12-08' }]
*/
```

**Comment by spencermountain** [comment]
hey thanks - yeah, let's tighten it up. ya, definitely the best outcome would be passing a config to compromise, on how best to handle ambiguous dates.

**Comment by spencermountain** [comment]
good point Howard - ya, the blurriness in the docs really demonstrates the lack of clarity in the project. As Frasier mentioned, a lot of the date-parsing is done with [spacetime](https://github.com/spencermountain/spacetime), which does an ok job at british formats - and this compromise-dates plugin could simply pass these british-dates through - it's very doable, and would be a welcome PR.

If I remember:
* 03/26 - march 26
* 26/03 - march 26
* 03/04 - march 4
* 03/04 + {dmy: true} -april 3rd

open to either of you, if you've got some spare cycles. I can put it on my list for the future, otherwise.
cheers

**Comment by hchiam** [comment]
feel free to put it in the to-do list, but i plan to sometime next week
investigate adding a dmy option to compromise-dates

File: plugins/dates/src/model/regex.js (hunk 1 above)
Comment (hchiam) [reply plugins/dates/src/model/regex.js:13]: https://github.com/spencermountain/compromise/pull/1133/files

**Pull request merged**